assets/src/data/eventespresso/core/actions/persist-entity-generators.js

```diff
@@ -280,11 +280,40 @@ function* persistAllDeletes() {
 	return { deleted: deletedIds, trashed: trashedIds };
 }
 
+function* persistAllEntities() {
+	const entitiesInState = yield select(
+		CORE_REDUCER_KEY,
+		'getAllEntitiesInState'
+	);
+	// first let's get a map of dirty entities to their models in a simple array.
+	// loop through the immutable Map and dispatch actions for dirty entities
+	const entitiesByModel = [];
+	entitiesInState.forEach( ( entityMap, modelName ) => {
+		const dirtyEntities = entityMap.filter( ( entity ) => entity.isDirty );
+		if ( dirtyEntities.length > 0 ) {
+			entitiesByModel.push( [ modelName, dirtyEntities ] );
+		}
+	} );
+	while ( entitiesByModel.length > 0 ) {
+		const [ modelName, entities ] = entitiesByModel.pop();
+		while ( entities.length > 0 ) {
+			const entity = entities.pop();
+			yield dispatch(
+				CORE_REDUCER_KEY,
+				'persistEntityRecord',
+				modelName,
+				entity
```

Comment:
maybe this could be 

```js
		while ( entities.length > 0 ) {
			yield dispatch(
				CORE_REDUCER_KEY,
				'persistEntityRecord',
				modelName,
				entities.pop()
			);
```

like you did below in `persistAllAddRelations()` ???

saves the creation of a variable...

Reply:
yup good spot 👍

Reply:
Actually, this is different because I'm using destructure assignment for the entities from the `entitiesByModel` array (which is an array of arrays).  So this needs to stay as is.

Reply:
just curious...

how does the creation of the `entity` variable affect what's happening in the `entitiesByModel` array?

regardless of whether you assign the return value from `entities.pop()` to the `entity` variable or just call `entities.pop()` directly within the dispatch, you are removing the last entity from the `entities` array (which has already been popped off the end of the `entitiesByModel` array).

Yes? No? Am I missing something?

I just want to understand how this is different.

Reply:
doh, sorry, I was basing my comment on the `entities` assignment from the `entitiesByModel.pop()` destructure.  For some reason (on this second look at things) I was focusing on the `entities` value instead of the `entity` value.  So ya I can push a change as you illustrated in your example.

Reply:
phew... thought I was going crazy... I could not for the life of me think of how things would be affected differently by my suggestion.
totally shook my understanding of JS (as limited as it already is).

Reply:
Ha! Sorry about that.  That was entirely my bad there.